File: 카카오/2021카카오블라인드_2.cpp
#include<bits/stdc++.h>

using namespace std;

bool compare(pair<string, int> a, pair<string, int> b) {
	return a.second < b.second;
}

vector<string> comb(string order, int size) {
	vector<string> ret;
	vector<int> v(order.size(), 0);

	sort(order.begin(), order.end());

	for (int i = 0; i < size; i++) {
		v[i] = 1;
	}

	do {
		string str = "";

		for (int i = 0; i < order.size(); i++) {
			if (v[i] == 1)
				str += order[i];
		}
		ret.push_back(str);
	} while (prev_permutation(v.begin(), v.end()));

	return ret;
}

vector<string> solution(vector<string> orders, vector<int> course) {
	vector<string> answer;

	for (int size : course) {
		map<string, int> map;

		for (string order : orders) {
			if (order.size() < size)
				continue;

			for (string key : comb(order, size)) {
				map[key]++;
			}
		}

		int max = (*max_element(map.begin(), map.end(), compare)).second;
		if (max < 2)
			max = 2;

		for (pair<string, int> p : map) {
			if (p.second == max)
				answer.push_back(p.first);
		}
	}
	sort(answer.begin(), answer.end());
	return answer;
}

int main() {
	vector<int> v(7);

	cout << v[4];
}

File: 카카오/2021카카오블라인드_2.java
import java.util.*;

class Solution {
    public String[] solution(String[] orders, int[] course) {
        List<String> answer = new ArrayList<>();
        Map<Integer, Map<String, Integer>> map = new HashMap<>();

        for (String order : orders) {
            char[] chars = order.toCharArray();
            Arrays.sort(chars);
            String sortedStr = new String(chars);

            for (int x : course) {
                Map<String, Integer> tmp = map.getOrDefault(x, new HashMap<>());
                combination(sortedStr, "", 0, x , tmp);

                if (!tmp.isEmpty())
                    map.put(x, tmp);
            }
        }

        for (int x : course) {
            if (map.containsKey(x)) {
                Map<String, Integer> tmp = map.get(x);
                int max = Math.max(2, Collections.max(tmp.values()));

                for (Map.Entry<String, Integer> entry : tmp.entrySet()) {
                    if (entry.getValue().equals(max)){
                        answer.add(entry.getKey());
                    }
                }
            }
        }

        Collections.sort(answer);
        String[] ans = new String[answer.size()];

        int idx = 0;
        for (String s : answer) {
            ans[idx++] = s;
        }
        return ans;
    }

    private void combination(String input, String result, int start, int size, Map<String, Integer> map) {
        if (result.length() == size) {
            map.put(result, map.getOrDefault(result, 0) + 1);
            return;
        }

        for (int i = start; i < input.length(); i++) {
            combination(input, result + input.charAt(i), i + 1, size, map);
        }
    }
}